vault/diagnose/diagnose_tls_test.go

```diff
@@ -0,0 +1,146 @@
+package diagnose
+
+import (
+	"testing"
+
+	"github.com/hashicorp/vault/command/server"
+	"github.com/hashicorp/vault/internalshared/configutil"
+	"github.com/hashicorp/vault/internalshared/listenerutil"
+	"github.com/hashicorp/vault/vault"
+)
+
+func setup(t *testing.T) *vault.Core {
+	serverConf := &server.Config{
+		SharedConfig: &configutil.SharedConfig{
+			Listeners: []*configutil.Listener{
+				{
+					Type:                          "tcp",
+					Address:                       "127.0.0.1:443",
+					ClusterAddress:                "127.0.0.1:8201",
+					TLSCertFile:                   "./certs/server.crt",
+					TLSKeyFile:                    "./certs/server.key",
+					TLSClientCAFile:               "./certs/rootca.crt",
+					TLSMinVersion:                 "tls11",
+					TLSRequireAndVerifyClientCert: true,
+					TLSDisableClientCerts:         true,
+				},
+				{
+					Type:                          "tcp",
+					Address:                       "127.0.0.1:443",
+					ClusterAddress:                "127.0.0.1:8201",
+					TLSCertFile:                   "./certs/server2.crt",
+					TLSKeyFile:                    "./certs/server2.key",
+					TLSClientCAFile:               "./certs/rootca2.crt",
+					TLSMinVersion:                 "tls12",
+					TLSRequireAndVerifyClientCert: true,
+					TLSDisableClientCerts:         false,
+				},
+				{
+					Type:                          "tcp",
+					Address:                       "127.0.0.1:443",
+					ClusterAddress:                "127.0.0.1:8201",
+					TLSCertFile:                   "./certs/server3.crt",
+					TLSKeyFile:                    "./certs/server3.key",
+					TLSClientCAFile:               "./certs/rootca3.crt",
+					TLSMinVersion:                 "tls13",
+					TLSRequireAndVerifyClientCert: false,
+					TLSDisableClientCerts:         true,
+				},
+			},
+		},
+	}
+	conf := &vault.CoreConfig{
+		RawConfig: serverConf,
+	}
+	core := vault.TestCoreWithConfig(t, conf)
+	return core
+}
+
+func TestTLSValidCert(t *testing.T) {
+	listeners := []listenerutil.Listener{
+		{
+			Config: &configutil.Listener{
+				Type:                          "tcp",
+				Address:                       "127.0.0.1:443",
+				ClusterAddress:                "127.0.0.1:8201",
+				TLSCertFile:                   "./../../api/test-fixtures/keys/cert.pem",
+				TLSKeyFile:                    "./../../api/test-fixtures/keys/key.pem",
+				TLSClientCAFile:               "./../../api/test-fixtures/root/rootcacert.pem",
+				TLSMinVersion:                 "0",
+				TLSRequireAndVerifyClientCert: true,
+				TLSDisableClientCerts:         false,
+			},
+		},
+	}
+	err := ListenerChecks(listeners)
+	if err != nil {
+		t.Fail()
```

Comment:
i think it would be helpful to fail with the error here. maybe a `t.Errorf()` with the error included?

Reply:
Done!